### PR TITLE
[15.0][MIG] product_sold_by_delivery_week: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36501

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@pedrobaeza please review

**Comment by cesar-tecnativa** [comment]
@carlosdauden please review

**Review by pedrobaeza — COMMENTED** [review]
There are no unit tests, so it's difficult to know if everything goes right.

**Comment by cesar-tecnativa** [comment]
> There are no unit tests, so it's difficult to know if everything goes right.

Added three tests

**Comment by cesar-tecnativa** [comment]
@LudLaf could you review this?

**Comment by pedrobaeza** [comment]
@LudLaf instead of editing from GitHub interface, you have to do it on your local and commit only once following commit message guidelines.

**Comment by LudLaf** [comment]
> @LudLaf instead of editing from GitHub interface, you have to do it on your local and commit only once following commit message guidelines.

Changes now made in local and pushed from there.

**Comment by cesar-tecnativa** [comment]
All the commits are correctly done now.

**Comment by cesar-tecnativa** [comment]
@LudLaf could you review this?

File: product_sold_by_delivery_week/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -1,12 +1,12 @@
-We can configure some variables with config parameter keys:
+Some variables can be configured using config parameter keys:
 
-- For the hint characters:
+- For hint characters:
 
   - For the sold char: `product_sold_by_delivery_week.sold_char`
   - For the not sold: `product_sold_by_delivery_week.not_sold_char`
-- For weeks length: `product_sold_by_delivery_week.weeks_to_consider`
+  - For weeks length: `product_sold_by_delivery_week.weeks_to_consider`
```
Comment (pedrobaeza) [inline product_sold_by_delivery_week/readme/CONFIGURE.rst]: Incorrect extra indentation level. This goes in the first indentation.

File: product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo.tests import TransactionCase
+
+
+class TestProductSoldByDeliveryWeek(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
```
Comment (pedrobaeza) [inline product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py]: Remove empty lines inside a method.

File: product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        cls.partner = cls.env.ref("base.res_partner_12")
```
Comment (pedrobaeza) [inline product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py]: Don't use demo data.

File: product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                "value": "M",
+            }
+        )
+        self.assertEqual(self.order.order_line[0].weekly_sold_delivered_shown, "MMMMMM")
+        self.assertEqual(self.order.order_line[1].weekly_sold_delivered_shown, False)
+
+    def test_03_sale_stock_delivery_partial(self):
+        """
+        Test a SO with a product on delivery.
+        """
```
Comment (pedrobaeza) [inline product_sold_by_delivery_week/tests/test_product_sold_by_delivery_week.py]:
```suggestion
        """Test a SO with a product on delivery."""
```

**Review by pedrobaeza — COMMENTED** [review]
It seems there's still a lot of untested code.

**Comment by sergio-teruel** [comment]
ping  @chienandalu

**Review by sergio-teruel — APPROVED** [review]

**Comment by rousseldenis** [comment]
@cesar-tecnativa Could you attend @pedrobaeza comments?

**Comment by pedrobaeza** [comment]
César is no longer working on Odoo world, so another person will attend it.

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!!!
ping @CarlosRoca13

**Comment by carlosdauden** [comment]
We should supersede and add changes on https://github.com/OCA/sale-reporting/pull/176

**Comment by CarlosRoca13** [comment]
I will add the commits when this PR https://github.com/OCA/sale-reporting/pull/196 will be marged to fix the CI too

**Comment by CarlosRoca13** [comment]
Done @carlosdauden

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**